File: ViewModels/HistoryViews/GetDetailsGameHistoryView.cs
using BJ.ViewModels.EnumsViews;
using System;
using System.Collections.Generic;

namespace BJ.ViewModels.HistoryViews
{
    public class GetDetailsGameHistoryView
    {
        public GetDetailsGameHistoryView()
        {
            Bots = new List<BotGetDetailsGameHistoryViewItem>();
        }

        public DateTime DateTime { get; set; }
        public string Winner { get; set; }
        public int CountBots { get; set; }

        public UserGetDetailsGameHistoryView User { get; set; }
        public List<BotGetDetailsGameHistoryViewItem> Bots { get; set; }

    }

    public class UserGetDetailsGameHistoryView
    {
        public UserGetDetailsGameHistoryView()
        {
            Cards = new List<StepUserGetDetailsGameHistoryViewItem>();
        }

        public string Name { get; set; }
        public List<StepUserGetDetailsGameHistoryViewItem> Cards { get; set; }

    }

    public class BotGetDetailsGameHistoryViewItem
    {
        public BotGetDetailsGameHistoryViewItem()
        {
            Cards = new List<StepBotGetDetailsGameHistoryViewItem>();
        }

        public string Name { get; set; }
        public List<StepBotGetDetailsGameHistoryViewItem> Cards { get; set; }
      
    }

    public class StepUserGetDetailsGameHistoryViewItem
    {
        public SuitTypeView Suit { get; set; }
        public RankTypeView Rank { get; set; }
    }
    public class StepBotGetDetailsGameHistoryViewItem
    {
        public SuitTypeView Suit { get; set; }
        public RankTypeView Rank { get; set; }
    }
}

File: BJ.WEB/ClientApp/src/app/shared/models/history-views/get-details-game-history-view.ts
import { RankTypeView } from '../enums-views/rank-type-view';
import { SuitTypeView } from '../enums-views/suit-type-view';

export class GetDetailsGameHistoryView {


    public dateTime: Date;
    public countBots: number;
    public winner: string;

    public user: UserGetDetailsGameHistoryView;
    public bots: Array<BotGetDetailsGameHistoryViewItem>;
   
    /**
     *
     */
    constructor() {
        this.bots = [];
    }
}

export class UserGetDetailsGameHistoryView {
    public name: string;
    public points: number;
    public cards: Array<StepUserGetDetailsGameHistoryViewItem>;

    constructor() {
        this.cards = [];
    }

}

export class StepUserGetDetailsGameHistoryViewItem {
    public suit: SuitTypeView;
    public rank: RankTypeView;
}

export class BotGetDetailsGameHistoryViewItem {
    public name: string;
    public points: number;
    public cards: Array<StepBotGetDetailsGameHistoryViewItem>;

    constructor() {
        this.cards = [];
    }
}

export class StepBotGetDetailsGameHistoryViewItem {
    public suit: SuitTypeView;
    public rank: RankTypeView;
}
